keadm/cmd/keadm/app/cmd/edge/joinbeta.go

```diff
@@ -64,43 +64,46 @@ func NewJoinBetaCommand() *cobra.Command {
 		Long:         edgeJoinBetaDescription,
 		Example:      edgeJoinBetaExample,
 		SilenceUsage: true,
-		PreRun: func(cmd *cobra.Command, args []string) {
+		PreRunE: func(cmd *cobra.Command, args []string) error {
 			step.Printf("Check KubeEdge edgecore process status")
 			running, err := util.IsKubeEdgeProcessRunning(util.KubeEdgeBinaryName)
 			if err != nil {
-				klog.Exitf("Check KubeEdge edgecore process status failed: %v", err)
+				return fmt.Errorf("check KubeEdge edgecore process status failed: %v", err)
 			}
 			if running {
-				klog.Exitln("EdgeCore is already running on this node, please run reset to clean up first")
+				return fmt.Errorf("EdgeCore is already running on this node, please run reset to clean up first")
 			}
 
 			step.Printf("Check if the management directory is clean")
 			if _, err := os.Stat(util.KubeEdgePath); err != nil {
 				if os.IsNotExist(err) {
-					return
+					return nil
```

Comment:
should we return err when the directory not exist?

Reply:
no, this is to ensure the dir /etc/kubeedge does not exist, 
only if the dir exsit, we will report an error